device/Microsoft.Azure.Devices.Client/Transport/RetryDelegatingHandler.cs

```diff
@@ -98,12 +101,9 @@ public RetryDelegatingHandler(IPipelineContext context)
                 var sendState = new SendMessageState();
                 await this.internalRetryPolicy.ExecuteAsync(() =>
                 {
-                    if (cancellationToken.IsCancellationRequested)
-                    {
-                        return Common.TaskConstants.Completed;
-                    }
+                    cancellationToken.ThrowIfCancellationRequested();
```

Comment:
>ThrowIfCancellationRequested [](start = 38, length = 28)

In case of cancellation, why should we throw?  This looks more like a normal operation than exception?  If cancellation is due to operation timeout.... the exception is throw from TaskHelper.WithTimout().

Reply:
This is the standard TPL pattern: if the operation timed out or was cancelled for other reasons through the cancellation token, the Task.Status should be set to "Cancelled" and not "Done".

Setting it to "Done" would make the application believe that the operation completed successfully (e.g. telemetry was sent successfully) which is not the case. 

An application can be written to avoid await/GetResult() and instead query its [Status](https://msdn.microsoft.com/en-us/library/system.threading.tasks.task.status(v=vs.110).aspx) which in today's implementation is incorrectly set to "Done".